src/helper/subscriber/waf.cpp

```diff
@@ -123,9 +123,70 @@ dds::result format_waf_result(dds::result::code code, std::string_view json) {
 
     return res;
 }
+
+DDWAF_LOG_LEVEL spdlog_level_to_ddwaf(spdlog::level::level_enum level)
+{
+    switch(level) {
+    case spdlog::level::trace:
+        return DDWAF_LOG_TRACE;
+    case spdlog::level::debug:
+        return DDWAF_LOG_DEBUG;
+    case spdlog::level::info:
+        return DDWAF_LOG_INFO;
+    case spdlog::level::warn:
+        return DDWAF_LOG_WARN;
+    case spdlog::level::err: [[fallthrough]];
+    case spdlog::level::critical:
+        return DDWAF_LOG_ERROR;
+    case spdlog::level::off: [[fallthrough]];
+    default:
+        break;
+    }
+    return DDWAF_LOG_OFF;
+}
+
+void log_cb(DDWAF_LOG_LEVEL level, const char* function, const char* file,
+    unsigned line, const char* message, uint64_t message_len)
+{
+    auto logger = spdlog::default_logger();
```

Comment:
I don't think this can ever return null... if it can there's something wrong with the shutdown order and this check may not even help if spdlog has already deallocated its `details::registry` instance.

Reply:
You're right!